Platform/IdavilleBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.c

```diff
@@ -1065,9 +1065,13 @@ ShadowFspmCodeInCache (
     //
     Temp = AllocateTemporaryMemory (0);
 
-    while (ImageBase < ImageMax) {
-      *Temp = *(volatile UINT8 *)(UINTN)ImageBase;
-      ImageBase += 64;
+    if (Temp != NULL) {
+      while (ImageBase < ImageMax) {
+        *Temp = *(volatile UINT8 *)(UINTN)ImageBase;
+        ImageBase += 64;
+      }
+    } else {
+      DEBUG ((DEBUG_ERROR, "Failed to Allocate temporary memory pool\n"));
     }
```

Comment:
This "else" logic is not required, or change to use DEBUG_INFO. As the comments mentioned memory allocation is just for GCC warning. it doesn't matter if there is failure.

Reply:
Updated with DEBUG_INFO